### PR TITLE
fix(loader-utils): ReadableFile implementation to match the interface

### PR DESCRIPTION
**Pull request by dariaterekhova-actionengine**
Changing input types of `ReadableFile` implementations to match the interface. Currently it's not matched and that cause runtime errors. I need this change because I'm currently working on local slpk reading for I3S explorer and I need `BlobFile` for it

### CODE REVIEW

File: modules/loader-utils/src/lib/files/blob-file.ts [hunk 1]
```diff
@@ -30,8 +30,10 @@ export class BlobFile implements ReadableFile {
     };
   }
 
-  async read(start: number, length: number): Promise<ArrayBuffer> {
-    const arrayBuffer = await this.handle.slice(start, start + length).arrayBuffer();
+  async read(start?: number | BigInt, length?: number): Promise<ArrayBuffer> {
+    const arrayBuffer = await this.handle
```
Comment (ibgreen) [inline modules/loader-utils/src/lib/files/blob-file.ts:34]: Why this change?

File: modules/loader-utils/src/lib/files/http-file.ts [hunk 2]
```diff
@@ -30,8 +30,8 @@ export class HttpFile implements ReadableFile {
     };
   }
 
-  async read(offset: number | bigint, length: number): Promise<ArrayBuffer> {
-    const response = await this.fetchRange(offset, length);
+  async read(offset?: number | bigint, length?: number): Promise<ArrayBuffer> {
+    const response = await this.fetchRange(offset ?? 0, length ?? 0);
```
Comment (ibgreen) [inline modules/loader-utils/src/lib/files/http-file.ts:34]: Use default argument syntax for better clarity?
```suggestion
  async read(offset: number | bigint = 0, length: number = 0): Promise<ArrayBuffer> {
    const response = await this.fetchRange(offset, length);
```

File: modules/loader-utils/src/lib/files/node-file-facade.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export class NodeFileFacade implements ReadableFile, WritableFile {
     throw new Error('Can\'t instantiate NodeFile. Make sure to import @loaders.gl/polyfills first.');
   }
   /** Read data */
-  async read(start?: number | bigint, end?: number | bigint): Promise<ArrayBuffer> {
+  async read(start?: number | bigint, offset?: number): Promise<ArrayBuffer> {
```
Comment (ibgreen) [inline modules/loader-utils/src/lib/files/node-file-facade.ts:28]: - What is the difference between `start` and `offset`? 
- How does this function now know where to stop reading?
- Should it be `length` instead?

File: modules/loader-utils/src/lib/files/blob-file.ts (hunk 1 above)
Comment (dariaterekhova-actionengine) [reply modules/loader-utils/src/lib/files/blob-file.ts:34]: I faced a problem when I have `ReadableFile` as input in `FileHandler` and read() there accepts `start` as `number | bigint` so I pass bigint here and got runtime error that it cannot sum bigint and number because real implementation knows nothing about bigint. We need keep real implementation of the methods matching it's description in `ReadableFile`

File: modules/loader-utils/src/lib/files/node-file-facade.ts (hunk 3 above)
Comment (dariaterekhova-actionengine) [reply modules/loader-utils/src/lib/files/node-file-facade.ts:28]: That's my mistake, I'll change it to `length`, as it's in the `ReadableFile` and `NodeFile` implementations

**Review by ibgreen — APPROVED** [review]
